pypika/terms.py

```diff
@@ -1259,7 +1259,10 @@ def get_function_sql(self, **kwargs: Any) -> str:
 
         return "{name}({args}{special})".format(
             name=self.name,
-            args=",".join(self.get_arg_sql(arg, **kwargs) for arg in self.args),
+            args=",".join(
+                p.get_sql(with_alias=False, subquery=True, **kwargs) if hasattr(p, "get_sql") else str(p)
```

Comment:
Can you please keep using self.get_arg_sql(arg, **kwargs) for the non subquery case as before? Its implementation isn't exactly equal to str(p).

Reply:
I think that this function `get_function_sql` is called if once this is a sub-query, but not all of the terms

Reply:
because function is only used in subquery

Reply:
Not sure I understood you. I mean changing:

`args=",".join(
                p.get_sql(with_alias=False, subquery=True, **kwargs) if hasattr(p, "get_sql") else str(p)` 
                
to:

`args=",".join(
                p.get_sql(with_alias=False, subquery=True, **kwargs) if hasattr(p, "get_sql") else self.get_arg_sql(arg, **kwargs)`

Reply:
what is `arg`? I can't see it

Reply:
You've changed arg to p, as you can see in diff. So that's essentially the element from the self.args, which we are iterating.

Reply:
Arg is a better variable name in this case btw.